hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1426,7 +1426,9 @@ private void listStatusFindKeyInTableCache(
         }
         OzoneFileStatus fileStatus = new OzoneFileStatus(
             cacheOmKeyInfo, scmBlockSize, !OzoneFSUtils.isFile(cacheKey));
-        cacheKeyMap.put(cacheKey, fileStatus);
+        cacheKeyMap.putIfAbsent(cacheKey, fileStatus);
+      } else if (cacheOmKeyInfo == null) {
```

Comment:
``` java
else if (cacheOmKeyInfo == null &&   && cacheKey.startsWith(startCacheKey)
          && cacheKey.compareTo(startCacheKey) >= 0) ) {
```
I think we should also check the key prefix as cacheKeyMap is only constructed for key with given prefix.

Reply:
> ```java
> else if (cacheOmKeyInfo == null &&   && cacheKey.startsWith(startCacheKey)
>           && cacheKey.compareTo(startCacheKey) >= 0) ) {
> ```
> 
> I think we should also check the key prefix as cacheKeyMap is only constructed for key with given prefix.

Thanks @sadanand48 . I have handled the suggested change.

Reply:
@sadanand48 As discussed, I am keeping all deleted key entries in cacheKeyMap and not going to filter based on startKey due to unpredictable nature of cache flush race condition, and because of filtering deleted entries from cache based on startKey, `org.apache.hadoop.ozone.om.TestKeyManagerImpl#testListStatusWithTableCacheRecursive` were failing pagination test consistently.

Reply:
Sure , looks like we need to keep track of all deleted keys after the startKey but there is no way to seek in the map. Anyways we can still load all deleted keys as it  is just a cache and is limited by cache size. Thanks for looking into it.